presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -25,12 +28,174 @@ namespace fs = std::experimental::filesystem;
 
 namespace facebook::presto {
 
+namespace {
+
+void checkIncomingSystemProperties(
+    const std::unordered_map<std::string, std::string>& values) {
+  static const std::unordered_set<std::string_view> kSupportedSystemProperties{
+      SystemConfig::kMutableConfig,
+      SystemConfig::kPrestoVersion,
+      SystemConfig::kHttpServerHttpPort,
+      SystemConfig::kHttpServerReusePort,
+      SystemConfig::kDiscoveryUri,
+      SystemConfig::kMaxDriversPerTask,
+      SystemConfig::kConcurrentLifespansPerTask,
+      SystemConfig::kHttpExecThreads,
+      SystemConfig::kHttpServerHttpsPort,
+      SystemConfig::kHttpServerHttpsEnabled,
+      SystemConfig::kHttpsSupportedCiphers,
+      SystemConfig::kHttpsCertPath,
+      SystemConfig::kHttpsKeyPath,
+      SystemConfig::kHttpsClientCertAndKeyPath,
+      SystemConfig::kNumIoThreads,
+      SystemConfig::kNumQueryThreads,
+      SystemConfig::kNumSpillThreads,
+      SystemConfig::kSpillerSpillPath,
+      SystemConfig::kShutdownOnsetSec,
+      SystemConfig::kSystemMemoryGb,
+      SystemConfig::kAsyncCacheSsdGb,
+      SystemConfig::kAsyncCacheSsdCheckpointGb,
+      SystemConfig::kAsyncCacheSsdPath,
+      SystemConfig::kAsyncCacheSsdDisableFileCow,
+      SystemConfig::kEnableSerializedPageChecksum,
+      SystemConfig::kUseMmapArena,
+      SystemConfig::kMmapArenaCapacityRatio,
+      SystemConfig::kUseMmapAllocator,
+      SystemConfig::kEnableVeloxTaskLogging,
+      SystemConfig::kEnableVeloxExprSetLogging,
+      SystemConfig::kLocalShuffleMaxPartitionBytes,
+      SystemConfig::kShuffleName,
+      SystemConfig::kHttpEnableAccessLog,
+      SystemConfig::kHttpEnableStatsFilter,
+      SystemConfig::kRegisterTestFunctions,
+      SystemConfig::kHttpMaxAllocateBytes,
+      SystemConfig::kQueryMaxMemoryPerNode,
+      SystemConfig::kEnableMemoryLeakCheck,
+      SystemConfig::kRemoteFunctionServerThriftPort,
+  };
+
+  std::stringstream supported;
+  std::stringstream unsupported;
+  for (const auto& pair : values) {
+    ((kSupportedSystemProperties.count(pair.first) != 0) ? supported
+                                                         : unsupported)
+        << "  " << pair.first << "=" << pair.second << "\n";
+  }
+  auto str = supported.str();
+  if (!str.empty()) {
+    LOG(INFO) << "STARTUP: Supported system properties:\n" << str;
+  }
+  str = unsupported.str();
+  if (!str.empty()) {
+    LOG(WARNING) << "STARTUP: Unsupported system properties:\n" << str;
+  }
+}
+
+void checkIncomingNodeProperties(
```

Comment:
What does checkIncomingSystemProperties mean? The implementation here seems to be a bit complicated. What's the purpose of doing so? Thanks!

Reply:
@xiaoxmeng 
It checks all loaded properties against the list of registered ones and creates two sets: supported and unsupported.
We only log them for now.
This is the 1st step to clean up our config files, so we only have supported properties.
Later we will start failing to start (like Presto Java) when an unsupported property is found in the prop files.